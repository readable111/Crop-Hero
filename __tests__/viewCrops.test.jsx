/****
 * @author Isaac Boodt
 * @reviewer
 * @tester 
 ***/

import { render } from "@testing-library/react-native";
import ViewCrops from '../app/viewcrops'
import * as Font from 'expo-font';
import AsyncStorage from '@react-native-async-storage/async-storage';

jest.mock('expo-font', ()=>({
        useFonts: jest.fn(),
        loadAsync: jest.fn().mockResolvedValue(true),
        isLoaded: jest.fn(()=>true)
}))

describe('<ViewCrops/>', () =>{
        beforeEach(async () => {
                await AsyncStorage.setItem("dark_mode_setting", false);
        
                Font.useFonts.mockReturnValue([true, false])
                
                const [fontsLoaded, fontError] = Font.useFonts()
                expect(fontsLoaded).toBe(true)
                expect(fontError).toBe(false)
        
        })

        test('renders correctly', () =>{
                const tree = render(<ViewCrops/>).toJSON();
                expect(tree).toMatchSnapshot();
        })
})