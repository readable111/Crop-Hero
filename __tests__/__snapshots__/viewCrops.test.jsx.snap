// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewCrops/> renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#F1DDBF",
          "borderColor": "#20232a",
          "borderWidth": 1,
          "fontSize": 42,
          "padding": 18,
          "textAlign": "right",
        },
        false,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "position": "relative",
        }
      }
    >
      <Text
        style={
          [
            {
              "alignContent": "center",
              "flex": 1,
              "fontFamily": "Domine-Medium",
              "fontSize": 42,
              "textAlign": "right",
            },
            false,
          ]
        }
      >
        View Crops
      </Text>
      <View
        style={
          {
            "backgroundColor": "#00DC40",
            "borderBottomLeftRadius": 140,
            "borderBottomRightRadius": 140,
            "height": 70,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": -19,
            "width": 140,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#97A5BF",
          "flex": 1,
          "height": "100%",
        },
        false,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "width": 40,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID=""
      >
        <Image
          source={1}
          style={{}}
        />
      </View>
    </View>
    <RCTScrollView
      data={
        [
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "05/06/2024",
            "hrfNum": "193242",
            "indoors": "No",
            "label": "Carrot",
            "location": "Greenhouse",
            "medium": "Hugel Mound",
            "name": "Carrot",
            "source": "Home Depot",
            "type": "Standard",
            "variety": "Standard",
            "visible": "visible",
            "yield": "none",
          },
          {
            "active": "N",
            "comments": "None",
            "datePlanted": "01/24/2022",
            "hrfNum": "945304",
            "indoors": "Yes",
            "label": "Cabbage",
            "location": "Outside",
            "medium": "Hugel Mound",
            "name": "Cabbage",
            "source": "Friend Recommendation",
            "type": "Standard",
            "variety": "Standard",
            "visible": "not visible",
            "yield": "large",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "11/13/2019",
            "hrfNum": "835242",
            "indoors": "Yes",
            "label": "Potato",
            "location": "Dump",
            "medium": "Hugel Mound",
            "name": "Potato",
            "source": "Farmer's market",
            "type": "Standard",
            "variety": "Standard",
            "visible": "visible",
            "yield": "medium",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999999",
            "indoors": "No",
            "label": "Tomato",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999991",
            "indoors": "No",
            "label": "Tomato2",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato2",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999998",
            "indoors": "No",
            "label": "Tomato3",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato3",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999997",
            "indoors": "No",
            "label": "Tomato4",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato4",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999996",
            "indoors": "No",
            "label": "Tomato5",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato5",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999995",
            "indoors": "No",
            "label": "Tomato6",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato6",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999994",
            "indoors": "No",
            "label": "Tomato7",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato7",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
          {
            "active": "Y",
            "comments": "None",
            "datePlanted": "08/30/2023",
            "hrfNum": "999993",
            "indoors": "No",
            "label": "Tomato8",
            "location": "Greenhouse #2",
            "medium": "Hugel Mound",
            "name": "Tomato8",
            "source": "Gathered",
            "type": "Standard",
            "variety": "Green",
            "visible": "not visible",
            "yield": "small",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Carrot
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Cabbage
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Potato
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato2
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato3
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato4
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato5
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato6
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFADA",
                    "borderColor": "#20232a",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "fontSize": 38,
                    "marginBottom": 20,
                    "marginHorizontal": 20,
                    "padding": 20,
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Domine-Regular",
                    "fontSize": 24,
                    "textAlign": "center",
                  }
                }
              >
                Tomato7
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
